adyenv6core/src/com/adyen/v6/facades/DefaultAdyenCheckoutFacade.java

```diff
@@ -473,7 +475,10 @@ public OrderData authorisePayment(final HttpServletRequest request, final CartDa
 
         RequestInfo requestInfo = new RequestInfo(request);
         requestInfo.setShopperLocale(getShopperLocale());
-
+        if (customer == null && isGuestUserTokenisationEnabled()) {
+            customer = checkoutCustomerStrategy.getCurrentUserForCheckout();
+            customer.setType(CustomerType.GUEST);
```

Comment:
Since line 479 is the same as 458, we can maybe avoid the if in lines 457-459?
```suggestion
        customer = checkoutCustomerStrategy.getCurrentUserForCheckout();
        if (checkoutCustomerStrategy.isAnonymousCheckout() && isGuestUserTokenisationEnabled()) {
            customer.setType(CustomerType.GUEST);
```

Reply:
It is intentional because if we will do move everything on line 459 then it will change the behaviour for RATEPAY. I want to keep other PMs intact.